python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -1121,10 +1094,15 @@ def getJobObject(self):
         Unknown: Why is this a GangaObject method and not a Node method?
         """
         from Ganga.GPIDev.Lib.Job import Job
-        r = self._getRoot(cond=lambda o: isinstance(o, Job))
-        if not isinstance(r, Job):
-            raise AssertionError('no job associated with object ' + repr(self))
-        return r
+        if self._getParent():
+            r = self._getRoot(cond=lambda o: isinstance(o, Job))
+            if not isinstance(r, Job):
+                raise AssertionError('No Job associated with object instead root=\'%s\' for \'%s\'' % (repr(self), type(r)))
+            return r
+        else:
+            if isinstance(self, Job):
+                return self
+        raise AssertionError('No Parent associated with object instead root=\'%s\' for \'%s\'' % (repr(self), str(None)))
```

Comment:
What issue is this change fixing?

Also, why on earth is this raising an `AssertionError`? There is no assertion being raised here. It should be a `RuntimeError` if anything.